code/game/machinery/turrets/turrets.dm

```diff
@@ -0,0 +1,225 @@
+	if(istype(internal_cell))
+		QDEL_NULL(internal_cell)
+	if(istype(installed_gun))
+		QDEL_NULL(installed_gun)
+	STOP_PROCESSING(SSfastprocess, src) // TODO
+	return ..()
+
+/obj/machinery/turret/Process()
+	process_power()
+
+	state_machine.evaluate()
+	var/decl/state/turret/turret_state = state_machine.current_state
+	if(turret_state.type == /decl/state/turret/turning) // TODO remove after test.
+		process_turning()
+	else if(turret_state.type == /decl/state/turret/engaging)
+		process_shooting()
+
+/obj/machinery/turret/proc/process_power()
+	// If the turret is dead or turned off, don't draw any power.
+	if((stat & BROKEN) || !internal_cell)
+		update_use_power(POWER_USE_OFF)
+		// TODO Consider making turret stop processing later.
+		return
+	
+	// Draw idle power from cell if the area has no regular power.
+	else if(stat & NOPOWER)
+		// See if the turret should lose power.
+		if(!internal_cell.check_charge(idle_power_usage * CELLRATE))
+			update_use_power(POWER_USE_OFF)
+		//	set_state(TURRET_DEPOWERED)
+			// TODO ditto
+			return
+		else
+			internal_cell.use(idle_power_usage * CELLRATE)
+
+	// Charge the cell if possible.
+	else if(can_charge_cell && !internal_cell.fully_charged())
+		var/power_to_give = active_power_usage * CELLRATE
+	//	var/multiplier = time_per_process() / SSobj.wait // This is used so when the turret is using SSfastprocess it doesn't get supercharged.
+		var/multiplier = SSfastprocess.wait / SSmachines.wait // TODO remove
+		internal_cell.give(power_to_give * multiplier)
+		update_use_power(POWER_USE_ACTIVE)
+	// Draw idle power from APC.
+	else
+		update_use_power(POWER_USE_IDLE)
+
+/obj/machinery/turret/proc/is_depowered()
+	return (stat & BROKEN) || !internal_cell || !internal_cell.check_charge(idle_power_usage * CELLRATE)
+
+/*
+/obj/machinery/turret/proc/time_per_process()
+	return speed_process ? SSfastprocess.wait : SSobj.wait
+*/
+
+
+/obj/machinery/turret/proc/process_turning()
+	var/distance_from_target_bearing = closer_angle_difference(current_bearing, target_bearing)
+	var/distance_this_step = min(calculate_turn_rate_per_process(), abs(distance_from_target_bearing)) 
+	var/adjustment = distance_this_step * SIGN(distance_from_target_bearing)
+
+	// This probably won't work if traverse is greater than 180.
+	// If someone smarter than me knows how to make that work then please tell me how.
+	if(!angle_within_traverse(current_bearing + adjustment))
+		adjustment = adjustment * -1
+
+	set_bearing(current_bearing + adjustment)
+
+/obj/machinery/turret/proc/calculate_turn_rate_per_process()
+	return turning_rate / (1 SECOND / SSfastprocess.wait)
+
+
+/obj/machinery/turret/proc/process_shooting()
+	if(installed_gun)
+		installed_gun.Fire(target, src)
+
+
+/obj/machinery/turret/update_icon()
+	if(!turret_stand)
+		turret_stand = image(icon, "turretCover")
+		turret_stand.layer = src.layer - 0.01
+		turret_stand.appearance_flags = KEEP_APART|RESET_TRANSFORM|TILE_BOUND|PIXEL_SCALE 
+		overlays += turret_stand
```

Comment:
Please use `add_overlay()` rather than directly manipulating overlays.

Reply:
My original code had `add_overlay()` but for some reason I thought Neb didn't have that, yet now it shows up in VSC, so I'll put them back.